audiomentations/core/composition.py

```diff
@@ -125,20 +143,119 @@ class OneOf(BaseCompose):
+        self.should_apply = random.random() < self.p
+        if self.should_apply:
+            if type(self.nbr_transforms) == tuple:
+                if self.nbr_transforms[1] is None:
+                    nbr_transforms_to_apply = random.randint(
+                        self.nbr_transforms[0], len(self.transforms)
+                    )
+                else:
+                    nbr_transforms_to_apply = random.randint(
+                        self.nbr_transforms[0], self.nbr_transforms[1]
+                    )
+            else:
+                nbr_transforms_to_apply = self.nbr_transforms
+            all_transforms_indexes = list(np.arange(len(self.transforms)))
+            self.transform_indexes = sorted(
+                random.sample(all_transforms_indexes, nbr_transforms_to_apply)
+            )
+        return self.transform_indexes
+
+    def __call__(self, *args, **kwargs):
+        if not self.are_parameters_frozen:
+            kwargs["apply_to_children"] = False
+            self.randomize_parameters(*args, **kwargs)
+
+        if self.should_apply:
+            if "apply_to_children" in kwargs:
+                del kwargs["apply_to_children"]
+
+            if issubclass(type(self.transforms[0]), BaseSpectrogramTransform):
+                if "magnitude_spectrogram" in kwargs:
+                    transformed_data = self.transforms[self.transform_indexes[0]](
+                        kwargs["magnitude_spectrogram"]
+                    )
+                else:
+                    transformed_data = self.transforms[self.transform_indexes[0]](
+                        args[0]
+                    )
+
+                for transform_index in self.transform_indexes[1:]:
+                    transformed_data = self.transforms[
+                        self.transform_indexes[transform_index]
+                    ](transformed_data)
+
+            if issubclass(type(self.transforms[0]), BaseWaveformTransform):
+                # get access to the sample rate
+                if "sample_rate" in kwargs:
+                    samples = kwargs["samples"] if "samples" in kwargs else args[0]
+                    sample_rate = kwargs["sample_rate"]
+                else:
+                    samples = args[0]
+                    sample_rate = args[1]
+
+                transformed_data = self.transforms[self.transform_indexes[0]](
+                    samples, sample_rate
+                )
+
+                for transform_index in self.transform_indexes[1:]:
+                    transformed_data = self.transforms[transform_index](
+                        transformed_data, sample_rate
+                    )
+            return transformed_data
+
+        if "samples" in kwargs:
+            return kwargs["samples"]
+        elif "magnitude_spectrogram" in kwargs:
+            return kwargs["magnitude_spectrogram"]
+        else:
+            return args[0]
+
+
+class OneOf(BaseCompose):
```

Comment:
In the future, OneOf could be a subclass of SomeOf 👍

Reply:
That was my initial thinking but I wasn't sure about how the randomization would work so I prefered to have this approach for a first version. I can change it now :)

Reply:
Yeah, it can be done in a future PR. It's not a hurry :)